### PR TITLE
fix(qwen3-vl): avoid scatter crash on TPU by vectorizing multimodal embedding merge and casting dtype

### PR DESCRIPTION
**Pull request by RexBearIU**
# Description

Fixes a TPU core halt during SFT training with Qwen3-VL models (`test_qwen3_multimodal_sft.sh`).

### Root Cause
1. In `_merge_mm_embeddings_inner` (`src/maxtext/multimodal/utils.py`), multimodal embeddings (`float32`) were inserted into text hidden states (`bfloat16`) via JAX `.at[target_pos, :].set(multimodal_embeddings)`.
2. Under JAX 0.11.0 with strict type promotion, the backward pass lowered this mixed-type in-place scatter into an XLA `scatter_custom_fusion` on TPU, triggering a fatal `RuntimeUnexpectedCoreHalt` sequencer halt at Step 0.

### Fix
- Cast `multimodal_embeddings` explicitly to `text_embeddings.dtype`.
- Vectorized the insertion using `jnp.where(mask_expanded, mm_aligned, text_embeddings)` with a cumulative sum mask constraint (`cumsum_mask <= multimodal_embeddings.shape[0]`), eliminating the XLA scatter operation.

# Verification
- **Unit Tests**: All 23 unit tests in `multimodal_utils_test.py` pass.
- **End-to-End TPU Verification (`mlperf-v5p` v5p-8)**:
  - Step 1 (JetStream installation): Succeeded.
  - Step 2 (Base checkpoint decode): Succeeded.
  - Step 3 (SFT native training on ChartQA, 2 steps): Completed steps 0 & 1 and successfully saved `checkpoints/1/items`.
  - Step 4 (Trained checkpoint decode): Succeeded loading `checkpoints/1/items` and generated expected predictions.
  - Full run exited with `EXIT_CODE=0`.

### CODE REVIEW

File: src/maxtext/multimodal/utils.py [hunk 1]
```diff
@@ -190,27 +190,19 @@ def _merge_mm_embeddings_inner(
...
-  first_pos_embedding = text_embeddings[0]
-
-  # Perform the insertion.
-  merged = text_embeddings.at[target_pos, :].set(multimodal_embeddings)
-
-  # Restore the first position's embedding, in case it was overwritten.
-  merged = merged.at[0].set(first_pos_embedding)
-
-  return merged
+  mask_bool = mask != 0
```
[the remark this answers is not in the document]
Comment (RexBearIU) [reply src/maxtext/multimodal/utils.py]: Applied suggestion to restrict  to the first  active multimodal positions.

**Review by hengtaoguo — APPROVED** [review]
Thanks for the fix! Looks like the crash was introduced/spotted because of the new JAX version?

**Review by SurbhiJainUSC — APPROVED** [review]
